sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentProducer.java

```diff
@@ -226,24 +229,46 @@ private Flux<DocumentProducerFeedResponse> splitProof(Flux<DocumentProducerFeedR
             // so this is resilient to split on splits.
             Flux<DocumentProducer<T>> replacementProducers = replacementRangesObs.flux().flatMap(
                     partitionKeyRangesValueHolder ->  {
-                        if (logger.isInfoEnabled()) {
-                            logger.info("Cross Partition Query Execution detected partition [{}] split into [{}] partitions,"
-                                + " last continuation token is [{}]. - Context: {}",
-                                feedRange,
-                                partitionKeyRangesValueHolder.v.stream()
-                                                               .map(ModelBridgeInternal::toJsonFromJsonSerializable)
-                                                               .collect(Collectors.joining(", ")),
-                                lastResponseContinuationToken,
-                                this.operationContextTextProvider.get());
+                        if (partitionKeyRangesValueHolder == null
+                            || partitionKeyRangesValueHolder.v == null
+                            || partitionKeyRangesValueHolder.v.size() == 0) {
+                            logger.error("Failed to find at least one child range");
+                            return Mono.error(new IllegalStateException("Failed to find at least one child range"));
+                        }
+
+                        if (partitionKeyRangesValueHolder.v.size() == 1) {
+                            // The feedRange is gone due to merge
+                            // we are going to continue drain the current document producer
+                            // Due to the feedRange does not cover full partition anymore, during populateHeaders, startEpk and endEpk headers will be added
+                            if (logger.isDebugEnabled()) {
+                                logger.info(
```

Comment:
`logger.debug` or `logger.isInfoEnabled()` 
We should make this consistent with the below logging as well.

Reply:
Good catch, updated